### PR TITLE
[INFRA] add citation.cff... and SPRING CLEANING !!!!!

### PR DESCRIPTION
**Pull request by Remi-Gau**
- [x] lint code with miss hit as much as possible (through datalad run) (meaning most changes ot .m files were done manually)
- [x] manual fix of non controversial changes to pacify miss hit
  - [x] function name renaming so it matches filename
  - [x] remove non ascii characters
- [x] fix some warnings and dead code in doc
- [x] lint docs/conf.py with black
- [x] modify miss hit config to make it NOT complain given the current status of the code base

and eventually...
...
...
...

- [x] add CITATION.cff (and a makefile to help validate it)

### CODE REVIEW

File: docs/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+_build/*
```
Comment (Remi-Gau) [inline docs/.gitignore:1]: by the way: why are the local builds of the doc not ignored?

if we want to have a "local" version, a PDF may be better, no?

Comment (Remi-Gau) [reply docs/.gitignore:1]: I am adding this gitignore.

But I am asking permission to nuke the _build from the repo.

**Comment by Remi-Gau** [comment]
BTW the read the docs build are failing. Let me know if you want me to have a look into this.

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+identifiers:
+  - description: "This is the collection of archived snapshots of all releases of fMRwhy"
+    type: doi
+    value: "10.5281/zenodo.4527532"
+
+contact:
+  - family-names: "Heunis"
+    given-names: "Stephan"
+    affiliation: "FIXME"
+    email: FIXME@FIXME.com
```
Comment (Remi-Gau) [inline CITATION.cff:32]: @jsheunis you and other contributors should probably update those FIXME fields

I grabbed the names from the all contributors file

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+.PHONY: help 
+
+help: ## Show what this Makefile can do
+	@python -c "$$PRINT_HELP_PYSCRIPT" < $(MAKEFILE_LIST)
+
+# may require a "pip install cffconvert" to run
+validate_cff: CITATION.cff
+	cffconvert --validate
+
```
Comment (Remi-Gau) [inline Makefile:26]: more as doc for developers in case they don't remember how to validate those files

File: CITATION.cff (hunk 2 above)
Comment (jsheunis) [reply CITATION.cff:32]: Thanks Remi!

File: docs/.gitignore (hunk 1 above)
Comment (jsheunis) [reply docs/.gitignore:1]: Probably just me not being attentive... I agree that we can kick it out.

Comment (Remi-Gau) [reply docs/.gitignore:1]: done

**Pull request closed without merging**